concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+       physicalExamPath LIKE "notes/Progress Notes/Physical Exam/Physical Exam/Neurologic/GCS/__")
+      AND (physicalexamoffset > 0 AND physicalexamoffset <= 1440 ) -- consider only first 24h
+      AND physicalexamvalue IS NOT NULL
+      AND patientunitstayid IS NOT NULL
+      )
+  GROUP BY patientunitstayid
+  )
+  AS physicalexam
+  ON physicalexam.patientunitstayid = pivoted_gcs.patientunitstayid
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND pivoted_gcs.gcs IS NOT NULL
+    AND pivoted_gcs.patientunitstayid IS NOT NULL 
+)
+
+
+-- Call merged_GCS table in one go
+, GCS_OASIS AS (
+    SELECT patientunitstayid AS pid_GCS
+    , CASE
+    WHEN MIN(gcs) < 8 THEN 10
+    WHEN MIN(gcs) BETWEEN 8 AND 13 THEN 4
+    WHEN MIN(gcs) = 14 THEN 3
+    WHEN MIN(gcs) = 15 THEN 0
+    ELSE NULL
+    END AS GCS_OASIS
+    FROM merged_GCS
+    --WHERE (chartoffset > 0 AND chartoffset <= 1440) -- already considered in step above
+    GROUP BY pid_GCS
+)
+
+-- Elective admission
+
+-- Mapping
+-- Assume emergency admission if patient came from
+-- Emergency Department, Direct Admit, Chest Pain Center, Other Hospital, Observation
+-- Assume elective admission if patient from other place, e.g. operating room, floor, etc.
+, elective_surgery AS (
+
+    -- 1: apachepredvar table
+    SELECT apache.patientunitstayid, electivesurgery2, adm_elective1, adm_elective2,
+    CASE 
+    WHEN electivesurgery = 1 THEN 1
+    ELSE NULL
+    END as electivesurgery1
+    FROM `physionet-data.eicu_crd.apachepredvar` AS apache
+
+    -- 2: admissiondx table
+    FULL OUTER JOIN(
+      
+      -- Concatenate string vars
+      WITH tt as (
+        SELECT patientunitstayid, string_agg(admitdxpath, ';') AS admit_text1, string_agg(admitdxname, ';') AS admit_text2, string_agg(admitdxtext, ';') AS admit_text3
+        FROM `physionet-data.eicu_crd.admissiondx`
+        GROUP BY patientunitstayid
+      )
+  
+    SELECT patientunitstayid, admit_text1, admit_text2, admit_text3
+    , CASE 
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|No%" THEN NULL
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|Yes%" 
+      AND ( (LOWER(admit_text1) LIKE "%surgery%" OR LOWER(admit_text2) LIKE "%surgery%" OR LOWER(admit_text3) LIKE "%surgery%")
+      OR  (LOWER(admit_text1) LIKE "%operative%" OR LOWER(admit_text2) LIKE "%operative%" OR LOWER(admit_text3) LIKE "%operative%") )
+      THEN 1
+    ELSE NULL
+    END AS electivesurgery2
+
+    , CASE 
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|No%" THEN NULL
+    WHEN admit_text1 LIKE "%admission diagnosis|Elective|Yes%" THEN 1
+    ELSE NULL
+    END AS adm_elective2
+
+    FROM tt
+    )
+    AS adm
+    ON adm.patientunitstayid = apache.patientunitstayid
+
+    -- 2: patient table
+    FULL OUTER JOIN (
+      SELECT patientunitstayid
+      , CASE
+      WHEN 
+        (FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC) LIKE "Emergency Department"
```

Comment:
In general, you shouldn't have to use a window function `FIRST_VALUE( ... )` with a `GROUP BY`. You use window functions when you don't want to collapse rows, but need to grab values across some group (like `patientunitstayid`). Conversely, if you want to collapse rows to one unique row per `patientunitstayid`, you use a `GROUP BY`.

In my view, you actually don't want to use either a window function or a group by. You just want a regular old case statement checking the `unitAdmitSource`. The statement `FIRST_VALUE(unitAdmitSource) OVER(ORDER BY unitAdmitSource DESC)` partitions the data by `ORDER BY unitAdmitSource DESC`, so it basically says "sort all the data by unitAdmitSource, then give me the first one. Since you haven't specified `patientunitstayid` in the partition, it won't consider it at all. My hunch is every single row will have the same value.